apps/studio/components/interfaces/Connect/DatabaseConnectionString.tsx

```diff
@@ -0,0 +1,512 @@
+  const emptyState = { db_user: '', db_host: '', db_port: '', db_name: '' }
+  const connectionInfo = pluckObjectFields(selectedDatabase || emptyState, DB_FIELDS)
+  const connectionTld =
+    projectDetails?.restUrl !== undefined
+      ? new URL(projectDetails?.restUrl ?? '').hostname.split('.').pop() ?? 'co'
+      : 'co'
+
+  const handleCopy = (id: string) => {
+    const labelValue = CONNECTION_TYPES.find((type) => type.id === id)?.label
+    sendEvent({
+      category: 'settings',
+      action: 'copy_connection_string',
+      label: labelValue ? labelValue : '',
+    })
+  }
+
+  const connectionStrings =
+    isSuccessPoolingInfo && poolingConfiguration !== undefined
+      ? getConnectionStrings(connectionInfo, poolingConfiguration, {
+          projectRef,
+        })
+      : {
+          direct: {
+            uri: '',
+            psql: '',
+            golang: '',
+            jdbc: '',
+            dotnet: '',
+            nodejs: '',
+            php: '',
+            python: '',
+            sqlalchemy: '',
+          },
+          pooler: {
+            uri: '',
+            psql: '',
+            golang: '',
+            jdbc: '',
+            dotnet: '',
+            nodejs: '',
+            php: '',
+            python: '',
+            sqlalchemy: '',
+          },
+        }
+
+  const poolerTld =
+    isSuccessPoolingInfo && poolingConfiguration !== undefined
+      ? getPoolerTld(poolingConfiguration?.connectionString)
+      : 'com'
+
+  // @mildtomato - Possible reintroduce later
+  //
+  // const poolerConnStringSyntax =
+  //   isSuccessPoolingInfo && poolingConfiguration !== undefined
+  //     ? constructConnStringSyntax(poolingConfiguration?.connectionString, {
+  //         selectedTab,
+  //         usePoolerConnection: snap.usePoolerConnection,
+  //         ref: projectRef as string,
+  //         cloudProvider: isProjectLoading ? '' : project?.cloud_provider || '',
+  //         region: isProjectLoading ? '' : project?.region || '',
+  //         tld: snap.usePoolerConnection ? poolerTld : connectionTld,
+  //         portNumber: `[5432 or 6543]`,
+  //       })
+  //     : []
+  // useEffect(() => {
+  //   // if (poolingConfiguration?.pool_mode === 'session') {
+  //   //   setPoolingMode(poolingConfiguration.pool_mode)
+  //   // }
+  // }, [poolingConfiguration?.pool_mode])
+
+  const lang = CONNECTION_TYPES.find((type) => type.id === selectedTab)?.lang ?? 'bash'
+  const contentType =
+    CONNECTION_TYPES.find((type) => type.id === selectedTab)?.contentType ?? 'input'
+
+  const example: Example | undefined = examples[selectedTab as keyof typeof examples]
+
+  const exampleFiles = example?.files
+  const exampleInstallCommands = example?.installCommands
+  const examplePostInstallCommands = example?.postInstallCommands
+  const hasCodeExamples = exampleFiles || exampleInstallCommands
+  const fileTitle = CONNECTION_TYPES.find((type) => type.id === selectedTab)?.fileTitle
+
+  let stepNumber = 0
```

Comment:
do we really need this? i see that we're assigning `number` property to `StepLabel`
can't we just assign them numbers as I want adhere to immutability as much as possible for our code practice